utility/unit_of_work/module.go

```diff
@@ -58,51 +58,55 @@ func (uow *baseUtilityUnitOfWork) ApplyBlock() error {
 		return coreTypes.ErrProposalBlockNotSet()
 	}
 
+	if err := uow.newSavePoint(); err != nil {
+		return err
+	}
+
 	// begin block lifecycle phase
 	log.Debug().Msg("calling beginBlock")
 	if err := uow.beginBlock(); err != nil {
 		return err
 	}
 
+	// processProposalBlockTransactions indexes the transactions into the TxIndexer.
+	// if it fails, it triggers a rollback to undo the changes that processProposalBlockTransactions
+	// could have caused.
 	log.Debug().Msg("processing transactions from proposal block")
 	txMempool := uow.GetBus().GetUtilityModule().GetMempool()
```

Comment:
`tl;dr` - I made `processProposalBlockTransactions` reference the bus to get access to the `Mempool` and now takes no argument.

Longer answer: Most of the discussion with @bryanchriswhite on this topic concluded with the idea that code should typically call the bus when it needs to access a module rather than take references to modules. Specifically for the function above, taking the `Mempool` from arguments as an interface makes it easy to test, but mocking a module on the bus is theoretically just as easy, yields the same flexibility, and still lets us abstract complexity behind interfaces.

Reply:
> st, but mocking a module on the bus is theoretically just as easy, yields the same flexibility, and still lets us abstract complexity behind interfaces.

Yup, that was the design goal. Mock this on the "dependency injection bus" and pass that around everywhere.